app/concerns/liquid_interpolatable.rb

```diff
@@ -132,6 +132,42 @@ def to_uri(uri, base_uri = nil)
       nil
     end
 
+    # Get the destination URL of a given URL by recursively following
+    # redirects, up to 5 times in a row.  If a given string is not a
+    # valid absolute HTTP URL, or any error occurs while following
+    # redirects, the original string is returned.
+    def uri_expand(url, limit = 5)
+      uri = URI(url)
+
+      http = Faraday.new do |builder|
+        builder.adapter :net_http
+        # builder.use FaradayMiddleware::FollowRedirects, limit: limit
+        # ...does not handle non-HTTP URLs.
+      end
+
+      limit.times do
+        begin
+          case uri
+          when URI::HTTP
+            response = http.head(uri)
+            case response.status
+            when 301, 302, 303, 307
+              if location = response['location']
+                uri += location
+                next
+              end
+            end
+          end
+        rescue
```

Comment:
There are too many kinds of errors that may be raised here, but I'll try to list them as many as I can think of.